### PR TITLE
feat(storefront) BCTHEME-381 add sufficient & informative text along with the color swatches

### PR DESCRIPTION
**Pull request by bc-alexsaiannyi**
#### What?

This PR adds setting to Page builder that provides/hides a text along with color swatches on  a Store. When it's set to 'Show' User'll see swatch name on picking an option. Screen Reader will announce it as well. All this will do Shopper Experience much better.
To achieve this handlebars logic has been divided on 2 parts:
-using dynamicComponent for all options except Swatches,
-add explicitly swatch component with flag that can be manipulate of  Show swatch name setting.

#### Tickets / Documentation
- [BCTHEME-381](https://jira.bigcommerce.com/browse/BCTHEME-381)

#### Screenshots (if appropriate)

https://user-images.githubusercontent.com/67792608/106276306-12563280-6240-11eb-97de-249d9ff2f142.mov

https://user-images.githubusercontent.com/67792608/106276322-17b37d00-6240-11eb-9480-d953b9a33a33.mov

https://user-images.githubusercontent.com/67792608/106276282-0a968e00-6240-11eb-8934-c6e30a88b893.mov

### CODE REVIEW

**Review by BC-tymurbiedukhin — COMMENTED** [review]
Could we put show_swatch_name setting into context and just add an additional change event handler for swatches?

**Comment by bc-alexsaiannyi** [comment]
> Could we put show_swatch_name setting into context and just add an additional change event handler for swatches?

yes, done

File: assets/js/theme/common/product-details.js [hunk 1]
```diff
@@ -17,11 +17,22 @@ export default class ProductDetails extends ProductDetailsBase {
         this.imageGallery = new ImageGallery($('[data-image-gallery]', this.$scope));
         this.imageGallery.init();
         this.listenQuantityChange();
+        this.swatchOptionMessageText = $('.swatch-option-message').text();
```
Comment (BC-tymurbiedukhin) [inline assets/js/theme/common/product-details.js:20]: Could we rename to swatchOptionMessageInitText?

File: assets/js/theme/common/product-details.js [hunk 2]
```diff
@@ -17,11 +17,22 @@ export default class ProductDetails extends ProductDetailsBase {
...
 
         const $form = $('form[data-cart-item-add]', $scope);
         const $productOptionsElement = $('[data-product-option-change]', $form);
         const hasOptions = $productOptionsElement.html().trim().length;
         const hasDefaultOptions = $productOptionsElement.find('[data-default]').length;
+        const $productSwatchGroup = $('[id*="attribute_swatch"]', $form);
+
+        if (context.showSwatchNames) {
+            $('.swatch-option-message').removeClass('u-hidden');
```
Comment (BC-tymurbiedukhin) [inline assets/js/theme/common/product-details.js:29]: Could we move $('.swatch-option-message') to class variable, we are using it at least 3 times

Comment (bc-alexsaiannyi) [reply assets/js/theme/common/product-details.js]: done

**Review by BC-tymurbiedukhin — APPROVED** [review]
LGTM

**Pull request merged**